RecoParticleFlow/PFClusterTools/src/PFEnergyCalibration.cc

```diff
@@ -24,117 +24,122 @@ PFEnergyCalibration::~PFEnergyCalibration()
-  faEtaBarrel->SetParameter(2,62.3599);
-  fbEtaBarrel = new TF1("fbEtaBarrelEH","[0]+[1]*exp(-x/[2])",1.,1000.);
-  fbEtaBarrel->SetParameter(0,0.0266189);
-  fbEtaBarrel->SetParameter(1,0.0828236);
-  fbEtaBarrel->SetParameter(2,68.0159);
-
-  // End-caps (fit made with eta 
-   faEndcap = new TF1("faEndcap","[0]+((([1]+([2]/x^[8]))*exp(-(x^[6]/[3])))-([4]*exp(-(x^[7]/[5]))))",1.,1000.);
-  faEndcap->SetParameter(0,3.10102);
-  faEndcap->SetParameter(1,37.0556);
-  faEndcap->SetParameter(2,-42.0353);
-  faEndcap->SetParameter(3,29.7847);
-  faEndcap->SetParameter(4,1.83545);
-  faEndcap->SetParameter(5,0.00670624);
-  faEndcap->SetParameter(6,0.89783);
-  faEndcap->SetParameter(7,-1.73037);
-  faEndcap->SetParameter(8,0.0329707);
-  fbEndcap = new TF1("fbEndcap","[0]+((([1]+([2]/x^[8]))*exp(-(x^[6]/[3])))-([4]*exp(-(x^[7]/[5]))))",1.,1000.);
-  fbEndcap->SetParameter(0,-1.07747);
-  fbEndcap->SetParameter(1,10.9055);
-  fbEndcap->SetParameter(2,-8.88263);
-  fbEndcap->SetParameter(3,0.0963555);
-  fbEndcap->SetParameter(4,-0.627371);
-  fbEndcap->SetParameter(5,0.0023847);
-  fbEndcap->SetParameter(6,-2.17868);
-  fbEndcap->SetParameter(7,-2.67134);
-  fbEndcap->SetParameter(8,-0.0108459);
-  fcEndcap = new TF1("fcEndcap","[0]+((([1]+([2]/x^[8]))*exp(-(x^[6]/[3])))-([4]*exp(-(x^[7]/[5]))))",1.,1000.);
-  fcEndcap->SetParameter(0,1.13795);
-  fcEndcap->SetParameter(1,1.21698);
-  fcEndcap->SetParameter(2,-3.81192);
-  fcEndcap->SetParameter(3,115.409);
-  fcEndcap->SetParameter(4,0.673456);
-  fcEndcap->SetParameter(5,0.217077);
-  fcEndcap->SetParameter(6,1.95596);
-  fcEndcap->SetParameter(7,-0.252215);
-  fcEndcap->SetParameter(8,0.506146);
-  faEtaEndcap = new TF1("faEtaEndcapEH","[0]+[1]*exp(-x/[2])",1.,1000.);
-  faEtaEndcap->SetParameter(0,0.0135773);
-  faEtaEndcap->SetParameter(1,-0.0276094);
-  faEtaEndcap->SetParameter(2,103.144);
-  fbEtaEndcap = new TF1("fbEtaEndcapEH","[0]+[1]*x*exp(x*[2])",1.,1000.);
-  fbEtaEndcap->SetParameter(0,0.181798);
-  fbEtaEndcap->SetParameter(1,-0.00087979);
-  fbEtaEndcap->SetParameter(2,-0.00231785);
- 
-  
+  //calibChrisClean.C calibration parameters shubham May 01, 2017
+  faBarrel = new TF1("faBarrel","[0]+((([1]+([2]/sqrt(x)))*exp(-(x^[6]/[3])))-([4]*exp(-(x^[7]/[5]))))",1.,1000.);
+  faBarrel->SetParameter(0,-13.9219);
+  faBarrel->SetParameter(1,14.9124);
+  faBarrel->SetParameter(2,5.38578);
+  faBarrel->SetParameter(3,0.861981);
+  faBarrel->SetParameter(4,-0.00759275);
+  faBarrel->SetParameter(5,3.73563e-23);
+  faBarrel->SetParameter(6,-1.17946);
+  faBarrel->SetParameter(7,-13.3644);
+  fbBarrel = new TF1("fbBarrel","[0]+((([1]+([2]/sqrt(x)))*exp(-(x^[6]/[3])))-([4]*exp(-(x^[7]/[5]))))",1.,1000.);
+  fbBarrel->SetParameter(0,2.18802);
+  fbBarrel->SetParameter(1,0.522053);
+  fbBarrel->SetParameter(2,-4.67354);
+  fbBarrel->SetParameter(3,1.2109e+06);
+  fbBarrel->SetParameter(4,1.75318);
+  fbBarrel->SetParameter(5,0.187919);
+  fbBarrel->SetParameter(6,-9241.43);
```

Comment:
Dear Slava,

Thank you very much for the review.

- here and for the barrel: which pages on the slides provided describe the magic numbers (as well as eta dependence change itself)? We have 1.3, 0.04, 0.05, and I didn't find a match in the slides in PPD

Some of these numbers were always present & needed to describe the eta dependence, we just made some optimizations.

- please rewrite this as

if (std::abs(eta)<2.5){
 etaCorrH = 1. + aEtaEndcapH(t) + 0.05*bEtaEndcapH(t);
} else {
etaCorrH = 1. + aEtaEndcapH(t) + bEtaEndcapH(t)*(0.04 + etaPow);
}

to avoid repeated calls

Also, please define auto absEta = std::abs(eta) and use it in this method instead of repeated calls to std::abs(eta)


: Done.


- is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

:Done

- were there any attempts made to check if all terms in the fit are meaningful?
exp( - x^{-9241}/1E6) looks like a 0

: Yes, we noticed a few cases and intended to fix these. Somehow didn't make it in this version of code. Also, would like to add that these functions here are meant for internal checks and not directly used in official workflow. Most of these details are important to fit the shape of calibrations in the lower energy regions.

I am pushing the changes to the code. I have reconstructed private QCD sample with these calibrations & final checks expected in a day or so, and PR can be merged then.